WasatchNET/MockSpectrometer.cs

```diff
@@ -0,0 +1,546 @@
+            get
+            {
+                return "MOCK";
+            }
+        }
+
+        public override uint integrationTimeMS
+        {
+            get
+            {
+                return integrationTimeMS_;
+            }
+            set
+            {
+                lock (acquisitionLock)
+                {
+                    //should logic for setting from stored data be here??
+                    integrationTimeMS_ = value;
+                }
+            }
+        }
+
+        public override bool laserEnabled // dangerous one to cache...
+        {
+            get
+            {
+                return laserEnabled_;
+            }
+            set
+            {
+                laserEnabled_ = value;
+            }
+        }
+
+        public override bool laserInterlockEnabled
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public override UInt64 laserModulationPeriod
+        {
+            get
+            {
+                return laserModulationPeriod_;
+            }
+            set
+            {
+                laserModulationPeriod_ = value;
+            }
+        }
+
+        public override UInt64 laserModulationPulseWidth
+        {
+            get
+            {
+                return laserModulationPulseWidth_;
+            }
+            set
+            {
+                laserModulationPulseWidth_ = value;
+            }
+        }
+
+        public override float laserTemperatureDegC
+        {
+            get
+            {
+                return 0;
+            }
+        }
+
+        public override ushort laserTemperatureRaw => 0;
+
+        public override byte laserTemperatureSetpointRaw
+        {
+            get
+            {
+                return laserTemperatureSetpointRaw_;
+            }
+            set
+            {
+                laserTemperatureSetpointRaw_ = Math.Min((byte)127, value);
+            }
+        }
+
+        public override ushort secondaryADC
+        {
+            get
+            {
+                return 0;
+            }
+        }
+
+        public override TRIGGER_SOURCE triggerSource
+        {
+            get
+            {
+                return triggerSource_;
+            }
+            set
+            {
+                triggerSource_ = value;
+            }
+        }
+
+        //////////////////////////////////////////////////////
+        //              NEEDS IMPLEMENT
+        //////////////////////////////////////////////////////
+        override internal bool open()
+        {
+            darkBaseline = 800;
+
+            eeprom = new EEPROM(this);
+
+            if (!eeprom.read())
+            {
+                logger.error("Spectrometer: failed to GET_MODEL_CONFIG");
+                //wrapper.shutdown();
+                close();
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool open(uint pixels)

```

Comment:
This is a good future state, and we should move towards it, but since I was using this in the context of WPSC I did not feel the need or want to replicate work that is already present there